Add the PWA Angular post (#1339)
GoogleChrome/web.dev: src/site/content/en/angular/creating-pwa-with-angular-cli/index.md

---
layout: post
title: "Create a Progressive Web App with the Angular CLI"
subhead: |
  You want to make your Angular app installable? Wait no more!
hero: installable.jpg
alt: Network cables.
date: 2019-08-15
description: |
  Learn how to create installable progressive Angular applications.
authors:
  - mgechev
tags:
  - angular
  - pwa
---

> In this post, you'll learn how to use the Angular command line interface (CLI) to make a [Progressive Web App](https://web.dev/discover-installable/) (PWA).

Reviewer: ```suggestion
In this post, you'll learn how to use the Angular command line interface (CLI) to make a [Progressive Web App (PWA)](/discover-installable).
```

_You can find the code sample from this guide [on GitHub](https://github.com/mgechev/manifest-web-dev)._

{% Aside %}

> This post assumes you're already familiar with PWAs and their benefits. If you need a refresher, check out the [Installable](https://web.dev/installable) collection.

Reviewer: ```suggestion
This post assumes you're already familiar with PWAs and their benefits. If you need a refresher, check out the [Installable](/installable) collection.
```

{% endAside %}

## Create an installable PWA

> To make your Angular application a PWA, all you need to do is run a single command!

Reviewer: ```suggestion
To make your Angular application a PWA, all you need to do is run a single command:
```

> 

Reviewer: ```suggestion
```

> 1. Run this command in the CLI:

Reviewer: ```suggestion
```
Sorry, I didn't catch in the doc that this kind of just repeats the previous sentence. Doesn't really seem like it's adding anything.

```bash
ng add @angular/pwa
```

This command will:

> 1. Create a [service worker](https://docs.google.com/document/d/1wj0NWqTmeLydh-EU3kH9orpVzXifQX6f4pjrsWo7Bv4/edit) with a default caching configuration.

Reviewer: ```suggestion
1. Create a [service worker](/precaching-with-the-angular-service-worker) with a default caching configuration.
```

Author: Ups, good catch!

> 2. Create a [manifest file](https://web.dev/add-manifest), which tells the browser how your app should behave when installed on the user's device.

Reviewer: ```suggestion
2. Create a [manifest file](/add-manifest), which tells the browser how your app should behave when installed on the user's device.
```

> 3. Adds a link to the manifest file in `index.html`.

Reviewer: ```suggestion
3. Add a link to the manifest file in `index.html`.
```

> 4. Adds the [`theme-color`](https://web.dev/themed-omnibox) `<meta>` tag to `index.html`.

Reviewer: ```suggestion
4. Add the [`theme-color`](/themed-omnibox) `<meta>` tag to `index.html`.
```

> 5. Creates app icons in the `src/assets` directory.

Reviewer: ```suggestion
5. Create app icons in the `src/assets` directory.
```

## Customize your PWA

> The [Precaching with the Angular service worker](https://web.dev/precaching-with-the-angular-service-worker/) post explains how to configure the Angular service worker. There you can find how to specify which resources you want the service worker to cache and what strategy it should use to do so.

Reviewer: ```suggestion
The [Precaching with the Angular service worker](/precaching-with-the-angular-service-worker) post explains how to configure the Angular service worker. There you can find how to specify which resources you want the service worker to cache and what strategy it should use to do so.
```

> The manifest file of your app lets you specify your app's name, short name, icons, theme color, and other details. Read about the full set of properties you can set on the MDN "[Web App Manifest](https://web.dev/add-manifest/)" page.

Reviewer: ```suggestion
The manifest file of your app lets you specify your app's name, short name, icons, theme color, and other details. Read about the full set of properties you can set on the [Add a web app manifest](/add-manifest/) post.
```

Take a peek at the manifest file generated by the Angular CLI:

```json
{
  "name": "manifest-web-dev",
  "short_name": "manifest-web-dev",
  "theme_color": "#1976d2",
  "background_color": "#fafafa",
  "display": "standalone",
  "scope": "/",
  "start_url": "/",
  "icons": [
    {
      "src": "assets/icons/icon-72x72.png",
      "sizes": "72x72",
      "type": "image/png"
    },
    …
    {
      "src": "assets/icons/icon-512x512.png",
      "sizes": "512x512",
      "type": "image/png"
    }
  ]
}
```

You can customize any of these properties by changing the relevant value in `manifest.webmanifest`.

> A PWA references its manifest file with a `link` element in `index.html`. Once the browser find" the reference it'll show the "Add to Home Screen" prompt:

Reviewer: ```suggestion
A PWA references its manifest file with a `link` element in `index.html`. Once the browser finds the reference, it'll show the **Add to Home screen** prompt:
```
